config/config_cmd.go

```diff
@@ -75,6 +103,92 @@ func redirectStdOut() (old, new *os.File) {
 	return
 }
 
+func printConfigTable(b interface{}, sectionName string, subsection bool, sectionKeyMap map[string]bool) {
+	val := reflect.Indirect(reflect.ValueOf(b))
+
+	if val.Kind() == reflect.Slice {
+		val = reflect.Indirect(reflect.ValueOf(val.Index(0).Interface()))
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Name", "Type", "Description"})
+	table.SetAlignment(3)
+	table.SetRowLine(true)
+
+	for i := 0; i < val.Type().NumField(); i++ {
+		t := val.Type().Field(i)
+		tagType := t.Type
+		if tagType.Kind() == reflect.Ptr {
+			tagType = t.Type.Elem()
+		}
+
+		fieldName := t.Name
+		fieldType := tagType.String()
+		fieldDescription := ""
+
+		if t.Type.Kind() == reflect.Struct {
+			ss := strings.Split(fieldType, ".")
+			fieldType = ss[len(ss)-1]
+		}
+
+		if jsonTag := t.Tag.Get("json"); jsonTag != "" && jsonTag != "-" {
+			var commaIdx int
+			if commaIdx = strings.Index(jsonTag, ","); commaIdx < 0 {
+				commaIdx = len(jsonTag)
+			}
+			fieldName = jsonTag[:commaIdx]
+		}
+
+		if pFlag := t.Tag.Get("pflag"); pFlag != "" && pFlag != "-" {
+			var commaIdx int
+			if commaIdx = strings.Index(pFlag, ","); commaIdx < 0 {
+				commaIdx = -1
+			}
+			fieldDescription = pFlag[commaIdx+1:]
+		}
+
+		if tagType.Kind() == reflect.Struct {
+			f := val.Field(i)
+			// In order to get value from unexported field
+			if f.Type().Kind() == reflect.Ptr {
+				f = reflect.NewAt(f.Type(), unsafe.Pointer(f.UnsafeAddr())).Elem()
+			} else {
+				f = reflect.NewAt(f.Type(), unsafe.Pointer(f.UnsafeAddr()))
+			}
+			if canPrint(f.Interface()) && sectionKeyMap[fieldType] == false {
+				sectionKeyMap[fieldType] = true
+				defer printConfigTable(f.Interface(), fieldType, true, sectionKeyMap)
+			}
+		}
+		if sectionKeyMap[fieldType] {
+			// Make this string as a link
+			fieldType = fmt.Sprintf("%s_", fieldType)
+		}
+
+		data := []string{fieldName, fieldType, fieldDescription}
+		table.Append(data)
+	}
+
+	// Print out the config table
+	fmt.Println(sectionName)
```

Comment:
re:
1. added it
2. There is a map to record the type that have been printed out, so that we will not get duplicate table
3. Fixed it. https://flyte--1608.org.readthedocs.build/en/1608/deployment/cluster_config/flytepropeller_config.html#section-plugins

Reply:
Awesome!